syntax = "proto3";

package game;

service Game {
    rpc MakeMove (Move) returns (EngineResponse);

    rpc IsWon (Message) returns (EngineResponse);
}

message Move {
    int32 row = 1;
    int32 col = 2;
    int32 subRow = 3;
    int32 subCol = 4;
    string player = 5;

}

message Message {
    int32 nothing = 1;
}

message EngineResponse{
    int32 won = 1;
    string player = 2;
}